cmd/kubeadm/app/preflight/checks.go

```diff
@@ -960,9 +960,15 @@ func RunJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.NodeConfigura
 		ServiceCheck{Service: "kubelet", CheckIfActive: false},
 		PortOpenCheck{port: 10250},
 		DirAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ManifestsSubDirName)},
-		FileAvailableCheck{Path: cfg.CACertPath},
 		FileAvailableCheck{Path: filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletKubeConfigFileName)},
 	}
+
+	// if the node is joining as worker node (not as a master)
+	if !cfg.Master {
+		// ensure /etc/kubernetes/pki/ca.crt does not exists
```

Comment:
exist

Reply:
what about a whole dir check? should there be any scenario where they're loading their own certs?

Reply:
I don't see any scenario apart HA which is in another branch of the code.
I will change this